org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/EclipseProjectImporter.java

```diff
@@ -32,17 +35,32 @@
 import org.eclipse.jdt.core.IJavaProject;
 import org.eclipse.jdt.core.JavaCore;
 import org.eclipse.jdt.ls.core.internal.AbstractProjectImporter;
+import org.eclipse.jdt.ls.core.internal.IProjectImporter;
 import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
+import org.eclipse.jdt.ls.core.internal.ProjectUtils;
 
 public class EclipseProjectImporter extends AbstractProjectImporter {
 
-	private Collection<java.nio.file.Path> directories;
-
 	@Override
 	public boolean applies(IProgressMonitor monitor) throws CoreException {
 		if (directories == null) {
 			BasicFileDetector eclipseDetector = new BasicFileDetector(rootFolder.toPath(), DESCRIPTION_FILE_NAME)
 					.addExclusions("**/bin");//default Eclipse build dir
+			List<IProjectImporter> importers = new ArrayList<>(ProjectsManager.importers());
+			for (IProject project : ProjectUtils.getAllProjects(false)) {
+				File projectFile = project.getLocation().toFile();
+				Iterator<IProjectImporter> iter = importers.iterator();
+				while (iter.hasNext()) {
+					IProjectImporter importer = iter.next();
+					if (!(importer instanceof EclipseProjectImporter)) {
+						importer.initialize(projectFile);
+						if (importer.applies(monitor)) {
```

Comment:
If the project is already resolved by other importer, should return early.

`if (importer.applies(folder) || importer.applies(monitor))`

Reply:
Fixed.